pkg/utils/authcheck/authtype.go

```diff
@@ -52,12 +55,15 @@ const (
 	BrokerServiceAccountName         = "broker"
 )
 
-var BrokerSecret = &corev1.SecretKeySelector{
-	LocalObjectReference: corev1.LocalObjectReference{
-		Name: "google-broker-key",
-	},
-	Key: "key.json",
-}
+var (
+	// Regex for a valid google service account email.
+	emailRegexp = regexp.MustCompile(`^[a-z][a-z0-9-]{5,29}@[a-z][a-z0-9-]{5,29}\.iam\.gserviceaccount.com$`)
```

Comment:
```suggestion
	emailRegexp = regexp.MustCompile(`^[a-z][a-z0-9-]{5,29}@[a-z][a-z0-9-]{5,29}\.iam\.gserviceaccount\.com$`)
```

As suggested by the bot.